### ISSUE TITLE
Wrong IE XP capabilities, results in false "protocol or cipher suite mismatch" and forces IIS admins to enable security holes

### ISSUE THREAD

**Issue by selecadm** [ssllabs/ssllabs-scan#89]
IE 8 / XP  Protocol or cipher suite mismatch (13.04.2014)
https://community.qualys.com/thread/12685

>The AllowInsecureRenegoClients is just to allow clients who do not support secure renegotiation

Protocol or cipher suite mismatch - IE 8 / XP (10.02.2015)
https://community.qualys.com/thread/14652

OP:
> It looks like when I allow insecure renegotiation, IE 8 / XP shows up as passing but that's obviously not a good solution.

Me:
> Visiting your site with IE 8 / XP is like clockwork.

https://www.ssllabs.com/ssltest/viewClient.html?name=IE&version=8&platform=XP
> Secure Renegotiation	No   INSECURE

Wow, really? Even IE 6 supports secure renegotiation by default.

IE 6 by default (SSLv2 ClientHello with SSL 3 support):
![image](https://cloud.githubusercontent.com/assets/10382329/6677599/9313920e-cc42-11e4-94c2-f7c05a5c1e50.png)

IE 6 with SSL 2 + SSL 3 + TLS 1.0 support (SSLv2 ClientHello):
![image](https://cloud.githubusercontent.com/assets/10382329/6677604/a3a56778-cc42-11e4-9ed5-3a3c94e9bcac.png)

IE 6 with SSL 2 only also supports secure renegotiation. Can you believe it?
![image](https://cloud.githubusercontent.com/assets/10382329/6677608/b884010e-cc42-11e4-8544-eec80cf95b24.png)

IE 8 by default:
![image](https://cloud.githubusercontent.com/assets/10382329/6677668/40584090-cc43-11e4-8bfb-57d8980a5c1a.png)

![image](https://cloud.githubusercontent.com/assets/10382329/6677631/ebede3ac-cc42-11e4-97e7-66da2ad0f903.png)

In conclusion, IE sends TLS_EMPTY_RENEGOTIATION_INFO_SCSV cipher suite when SSL 2 is enabled and/or TLS 1.0 is disabled. Otherwise, it sends renegotiation extension.

**Comment by ivanr** [comment]
If I understand your submission correctly, you're saying that patched XP does support secure renegotiation, whereas SSL Labs claims it does not?

**Comment by selecadm** [comment]
Yes, and this is results in "Protocol or cipher suite mismatch" in case of IIS server without AllowInsecureRenegoClients.

**Comment by selecadm** [comment]
@ivanr I have just had a dream about this issue, your message was that this is not so important and there are other things to consider. I hope that it won't be so :D.

**Comment by ivanr** [comment]
Well, it's true that it's not very important. The time spent on it might be better spent adding new features to SSL Labs. Do you think that users who are still using Windows XP are actually getting patches (or applying them) and that their browsers support secure renegotiation? I might change it still.

**Comment by selecadm** [comment]
Well, the patch was released 4,5 years ago, and there are servers that don't speak to non-RFC5746 compliant clients, so this patch is essential for users.
The current situation is confusing and forces admins to allow insecure clients, though they don't need to.

**Comment by ivanr** [comment]
Fixed in 1.16.1, now running on dev.ssllabs.com.
